File: quests/quest00/ex01.js
"use strict";
// my_first_file_with_content must exist and weigh 40 bytes. 2 points.

const FILE = "my_first_file_with_content";

exports.run = function (g) {
  const exists = g.check(
    `${FILE} exists and is a regular file`,
    g.isFile(FILE),
    g.exists(FILE) ? "" : "no such file in this directory"
  );

  const size = exists ? g.size(FILE) : null;
  g.check(
    `${FILE} size is 40`,
    size === 40,
    size === 40 ? "" : `size is ${size === null ? "unknown" : size}`
  );
};

File: quests/quest00/ex01.py
"""ex01 - my_first_file_with_content must exist and weigh 40 bytes. 2 points."""

FILE = "my_first_file_with_content"


def run(g):
    exists = g.check("%s exists and is a regular file" % FILE,
                     g.is_file(FILE),
                     "" if g.exists(FILE) else "no such file in this directory")

    size = g.size(FILE) if exists else None
    g.check("%s size is 40" % FILE,
            size == 40,
            "" if size == 40 else "size is %s" % ("unknown" if size is None else size))
